swift/ql/lib/codeql/swift/elements/decl/TypeDecl.qll

```diff
@@ -19,27 +19,53 @@ private import swift
 class TypeDecl extends Generated::TypeDecl {
   override string toString() { result = this.getName() }
 
+  /**
+   * Gets the `index`th base type of this type declaration (0-based).
+   *
+   * This is the same as `getImmediateInheritedType`.
+   * DEPRECATED: either use `getImmediateInheritedType` or unindexed `getABaseType`.
+   */
+  deprecated Type getImmediateBaseType(int index) { result = this.getImmediateInheritedType(index) }
+
+  /**
+   * Gets the `index`th base type of this type declaration (0-based).
+   * This is the same as `getInheritedType`.
+   * DEPRECATED: use `getInheritedType` or unindexed `getABaseType`.
+   */
+  deprecated Type getBaseType(int index) { result = this.getInheritedType(index) }
+
+  /**
+   * Gets any of the base types of this type declaration.
+   */
+  Type getABaseType() {
+    // TODO generalize this to resolve `TypeAliasDecl`s and consider bases added by extensions
```

Comment:
I'm happy for this TODO to be merged as I intend to follow-up with my own work that will resolve it.